akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -408,7 +410,17 @@ import akka.util.ByteString
       case OK =>
         result.getHandshakeStatus match {
           case NEED_WRAP =>
-            flushToUser()
+            // https://github.com/akka/akka/issues/29922
+            // A second workaround for an infinite loop we have not been able to reproduce/isolate,
+            // if you see this, and can reproduce consistently, please report back to the Akka team
+            // with a reproducer or details about the client causing it
+            unwrapPutBackCounter += 1
+            if (unwrapPutBackCounter > 1000) {
+              throw new IllegalStateException(
+                s"Stuck in unwrap loop, bailing out, last handshake status [$lastHandshakeStatus], " +
+                s"remaining=${transportInBuffer.remaining}, out=${userOutBuffer.position()}, " +
+                "(https://github.com/akka/akka/issues/29922)")
+            }
             transportInChoppingBlock.putBack(transportInBuffer)
           case FINISHED =>
             flushToUser()
```

Comment:
Comparing main and 2.6 I see that there is another change that might be related but I don't know. https://github.com/akka/akka/pull/31433/files

Reply:
Yeah, wasn't sure that was related, so opted to not backport that when I was looking at it. Not sure that is the right decision though, that flush-change is also pretty minimal.